build.c

```diff
@@ -199,6 +202,46 @@ buildadd(struct node *n)
 	e->flags &= ~FLAG_CYCLE;
 }
 
+static void
+statusfmt(char *fmt)
+{
+	struct timespec ctime;
+
+	for (; *fmt; ++fmt)
+		if (*fmt == '%')
+			switch (*(++fmt)) {
+			case 's':
+				printf("%zu", nstarted);
+				break;
+			case 'f':
+				printf("%zu", nfinished);
+				break;
+			case 't':
+				printf("%zu", ntotal);
+				break;
+			case 'r':
+				printf("%zu", nstarted - nfinished);
+				break;
+			case 'u':
+				printf("%zu", ntotal - nstarted);
+				break;
+			case 'p':
+				printf("%3lu%%", 100 * nfinished / ntotal);
+				break;
+			case 'e':
+				clock_gettime(CLOCK_REALTIME, &ctime);
```

Comment:
I think `CLOCK_MONOTONIC` is the right clock to use here. Then it won't be affected by a system clock change during the job (unlikely, but still).